visualizer/utils/button.py

```diff
@@ -0,0 +1,277 @@
+import pygame
+import math
+from visualizer.utils.text import Text
+from game.utils.vector import Vector
+from typing import Optional, Callable
+from typing import TypeAlias
+
+# Typing alias for color
+Color: TypeAlias = str | int | tuple[int, int, int, Optional[int]] | list[
+    int, int, int, Optional[int]] | pygame.Color
+
+
+# Class for colors used for button class
+class ButtonColors:
+    """
+    Class for using colors for the Button class
+
+    Can select three colors for both the text and button: default, mouse hover, and mouse clicked
+
+    Parameters:
+    fg_color           :  Used to store default text color              Default - #daa520
```

Comment:
It's good that the hex codes are here but add the actual color names next to the defaults to make it easier